pkg/model/gcemodel/network.go

```diff
@@ -35,35 +35,58 @@ var _ fi.ModelBuilder = &NetworkModelBuilder{}
 
 func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	network := &gcetasks.Network{
-		Name:      s(b.NameForNetwork()),
+		Name:      b.LinkToNetwork().Name,
 		Lifecycle: b.Lifecycle,
 		Mode:      "auto", // Automatically create subnets, but stop using legacy mode
+		Shared:    fi.Bool(true),
```

Comment:
This means new clusters will always run in an existing network. Should we have this mimic the behavior in AWS where we create new networks by default unless the ID of an existing one is provided?